blockchains/erc20/erc20_worker.js

```diff
@@ -37,11 +37,12 @@ class ERC20Worker extends BaseWorker {
     let events = [];
     let overwritten_events = [];
     if ('extract_exact_overwrite' === constants.CONTRACT_MODE) {
-      events = await contractEditor.getPastEventsExactContracts(this.web3, result.fromBlock, result.toBlock);
+      events = await contractEditor.getPastEventsExactContracts(this.web3, result.fromBlock, result.toBlock,
+        new TimestampsCache());
```

Comment:
Provide a cache which is intended to live just for the current set of events. The next set of events would be for new blocks so no need to persist it.